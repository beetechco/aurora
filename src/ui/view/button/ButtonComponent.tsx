
import * as React from "react";

interface IButtonComponentState {
  label: string;
}

interface IButtonComponentProps {
  id: string;
  label: string;
  onAction: Function;
}

export class ButtonComponent extends React.Component<
  IButtonComponentProps,
  IButtonComponentState
  > {
  constructor(props: any) {
    super(props);
    this.state = {
      label: this.props.label,
    };
  }

  changeLabel = (label: string): void => {
    this.setState({
      label,
    });
  };

  onClick = () => {
    if (this.props.onAction) {
      this.props.onAction();
    }
  };

  render(): any {
    return (
      <button
        id={`${this.props.id}:button`}
        onClick={this.onClick}
      >
        {this.state.label}
      </button>
    );
  }
}