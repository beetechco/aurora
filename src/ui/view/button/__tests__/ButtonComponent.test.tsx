import * as React from 'react';
import { shallow } from 'enzyme';
import { ButtonComponent } from '../ButtonComponent';

it('should change label', () => {
  let label = 'label';
  const buttonRef: any = React.createRef();
  const buttonComponent = shallow(
    <ButtonComponent
      id='testButton'
      ref={buttonRef}
      label={label}
      onAction={() => {}
      }
    />);

  expect(buttonComponent.text()).toBe(label);

  // change label
  label = 'newLabel';
  (buttonComponent.instance() as ButtonComponent).changeLabel(label);

  expect(buttonComponent.text()).toBe(label);

  expect(buttonComponent).toMatchSnapshot();
});

it('should call an action on click', () => {
  const label = 'label';
  const buttonRef: any = React.createRef();
  const onAction = jest.fn();

  const buttonComponent = shallow(
    <ButtonComponent
      id='testButton'
      ref={buttonRef}
      label={label}
      onAction={onAction}
    />);

  buttonComponent.find('button').simulate('click', { preventDefault() {} });

  expect(onAction.mock.calls.length).toBe(1);    
  expect(buttonComponent).toMatchSnapshot();
});