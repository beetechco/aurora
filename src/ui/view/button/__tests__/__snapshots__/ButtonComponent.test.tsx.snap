// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call an action on click 1`] = `
<button
  id="testButton:button"
  onClick={[Function]}
>
  label
</button>
`;

exports[`should change label 1`] = `
<button
  id="testButton:button"
  onClick={[Function]}
>
  newLabel
</button>
`;
